### DISCUSSION TITLE
Add simple icon on each file in repository

### DISCUSSION THREAD

**Discussion by davidleitw** [community/community#17951] [Other Feature Feedback, Questions, & Ideas]
Maybe github can add simple icon on each file in repository，that allows people to quickly know the type of file.

If someone don't like see many icon, they can set the icon back to default freely.

**Comment by TimNekk** [comment]
I think this will overload the minimalistic design. Currently, there is already a separation of icons into files and folders. And you can easily determine the file type just by looking at its extension.

In addition, there are a huge number of different file types. I guess creating a pack of icons for each type so that they fit into the design of github is very difficult.

**Reply by davidleitw** [reply]
That's why I thought it might be possible to let users customize their own themes，but i think you're right, it maybe a little breaking the minimalistic design。
